convert2rhel/systeminfo.py

```diff
@@ -397,10 +413,58 @@ def corresponds_to_rhel_eus_release(self):
         For example if we detect CentOS Linux 8.4, the corresponding RHEL 8.4 is an EUS release, but if we detect
         CentOS Linux 8.5, the corresponding RHEL 8.5 is not an EUS release.
 
-        :return: Whether or not the current system has a EUS correspondent in RHEL.
+        :return: Whether or not the current system has an EUS correspondent in RHEL.
         :rtype: bool
         """
-        return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+        eus_release_date = EUS_MINOR_VERSIONS.get("%s.%s" % (self.version.major, self.version.minor), False)
+        if eus_release_date:
+            eus_release_date = dt.strptime(eus_release_date, "%Y-%m-%d")
+            current_datetime = dt.today()
+            if not self.has_internet_access:
+                return self.datetime_comparison_returns(current_datetime, eus_release_date)
+            elif current_datetime > eus_release_date:
+                try:
+                    latest_release_version = self.get_latest_distro_release_version()
+                except RepoqueryFailure:
+                    return self.datetime_comparison_returns(current_datetime, eus_release_date)
+
+                latest_major, latest_minor = latest_release_version.split(".")
+                if int(latest_minor) > int(self.version.minor):
+                    return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+                if int(latest_minor) < int(self.version.minor):
+                    pass
+                    # corner case that should not happen, need to decide what to do in case we land in this condition
+                    # where the users system on a version that hasn't been released yet
+        return False
+
+    def datetime_comparison_returns(self, current_datetime, eus_release_date):
+        if current_datetime > eus_release_date:
+            return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+        return False
+
+    def get_latest_distro_release_version(self):
+        """Extract minor version from release name using yum list"""
+        cmd = [
+            "repoquery",
+            "--setopt=exclude=",
```

Comment:
Do we really need to override the exclude flag with this repoquery call? It seems to me a bit odd... In your tests, @pr-watson, have you encountered a case where this was needed?

Reply:
I saw it used with the query in is_loaded_kernel_latest so I figured i'd stick with the same format, we can remove it

Reply:
Well.. If for some reason the user specified that the `centos-release` package is excluded, then with this option we are overriding it and saying that it is not excluded anymore. 

I don't ever recall a case where I saw that it was excluded for some reason (The only one I can remember of top of my head was a kernel one). 

I would say that we can remove this part of the repoquery args, but if you want to check out with @bocekm or @abadger first, that would be good too.

Reply:
It makes sense to keep the `--setopt=exclude=` here. If a user has `exclude=centos-release` in `/etc/yum.conf` this repoquery call would fail. I see no reason to adhere to the custom user exclusion here.

Related: https://github.com/oamg/convert2rhel/pull/641

Reply:
Then lets leave it there